compiler/src/typed/env.re

```diff
@@ -1992,105 +2010,33 @@ let rec add_signature = (sg, env) =>
+let add_components = (slot, root, env0, comps) => {
   let add_l = (w, comps, env0) => TycompTbl.add_open(slot, w, comps, env0);
 
   let add = (w, comps, env0) => IdTbl.add_open(slot, w, root, comps, env0);
 
-  let skipped_modules = ref(StringSet.empty);
-  let filter = (tbl, env0_tbl) =>
-    switch (filter_modules) {
-    | None => tbl
-    | Some(f) =>
-      Tbl.fold(
-        (m, x, acc) =>
-          if (f(m)) {
-            Tbl.add(m, x, acc);
-          } else {
-            assert(
-              switch (IdTbl.find_name(m, env0_tbl, ~mark=false)) {
-              | _ => false
-              | exception _ => true
-              },
-            );
-            skipped_modules := StringSet.add(m, skipped_modules^);
-            acc;
-          },
-        tbl,
-        Tbl.empty,
-      )
-    };
-
-  let filter_and_add = (w, comps, env0) => {
-    let comps = filter(comps, env0);
-    add(w, comps, env0);
-  };
-
-  let filtered_components =
-    switch (filter_components) {
-    | Some(f) =>
-      let filter = tbl => {
-        let new_tbl = ref(Tbl.empty);
-        Tbl.iter(
-          (name, value) =>
-            switch (f(name)) {
-            | Some(new_name) => new_tbl := Tbl.add(new_name, value, new_tbl^)
-            | None => ()
-            },
-          tbl,
-        );
-        new_tbl^;
-      };
-
-      {
-        ...comps,
-        comp_constrs: filter(comps.comp_constrs),
-        comp_values: filter(comps.comp_values),
-        comp_types: filter(comps.comp_types),
-        comp_modtypes: filter(comps.comp_modtypes),
-        comp_components: filter(comps.comp_components),
-      };
-    | None => comps
-    };
-
   let constructors =
-    add_l(
-      x => `Constructor(x),
-      filtered_components.comp_constrs,
-      env0.constructors,
-    );
+    add_l(x => `Constructor(x), comps.comp_constrs, env0.constructors);
 
-  let values =
-    add(x => `Value(x), filtered_components.comp_values, env0.values);
+  let labels = add_l(x => `Label(x), comps.comp_labels, env0.labels);
 
-  let types = add(x => `Type(x), filtered_components.comp_types, env0.types);
+  let values = add(x => `Value(x), comps.comp_values, env0.values);
+
+  let types = add(x => `Type(x), comps.comp_types, env0.types);
 
   let modtypes =
-    add(
-      x => `Module_type(x),
-      filtered_components.comp_modtypes,
-      env0.modtypes,
-    );
+    add(x => `Module_type(x), comps.comp_modtypes, env0.modtypes);
 
   let components =
-    filter_and_add(
-      x => `Component(x),
-      filtered_components.comp_components,
-      env0.components,
-    );
+    add(x => `Component(x), comps.comp_components, env0.components);
 
-  let modules =
-    filter_and_add(
-      x => `Module(x),
-      filtered_components.comp_modules,
-      env0.modules,
-    );
+  let modules = add(x => `Module(x), comps.comp_modules, env0.modules);
 
   {
-    ...env0,
     summary: Env_open(env0.summary, root),
+    local_constraints: env0.local_constraints,
```

Comment:
Just for my own education, what is a `local_constraint` in the ocaml type system?

Reply:
I believe this is for like when you want to support pattern matching on an inline record constructor and then binding on that record and using it as a standalone value, but you can't let that type escape that local scope.

Reply:
Interesting! Thanks 😄